src/crimson/osd/ops_executer.cc

```diff
@@ -740,96 +741,128 @@ version_t OpsExecuter::get_last_user_version() const
   return pg->get_last_user_version();
 }
 
-void OpsExecuter::make_writeable(std::vector<pg_log_entry_t>& log_entries)
+// a counterpart of the `make_writeable()` in the classical OSD.
+// the main difference is this one executes before, not after,
+// all other operations. the rationale behind the departure is
+// simplicity (no new / old obs, no txn reordering).
+//
+// TODO: make this `static`
```

Comment:
Comparisons to classic are somewhat useful to some of us at the moment, but that won't last and it's not so good for anyone who isn't already familiar with classic.  Let's strive for documentation that describes what the function does rather than how it differs from something in classic (classic isn't known for functions with well-explained mechanics...).  Add something like this in the header:
```
/**
 * execute_clone
 * 
 * If snapc contains a snap which occurred logically after the last write
 * seen by this object (see OpsExecutor::should_clone()), we first need
 * make a clone of the object at its current state.  execute_clone primes
 * txn with that clone operation and returns an
 * OpsExecutor::CloningContext which will allow us to fill in the corresponding
 * metadata and log_entries once the operations have been processed.
 * 
 * Note that this strategy differs from classic, which instead performs this
 * work at the end and reorders the transaction.  See
 * PrimaryLogPG::make_writeable
 * 
 * @param snapc [in] snapc for this operation (from the client if from the
 *                                client, from the pool otherwise)
 * @param initial_obs [in] objectstate for the object at operation start
 * @param initial_snapset [in] snapset for the object at operation start
 * @param backend [in,out] interface for generating mutations
 * @param txn [out] transaction for the operation
 */
```

Reply:
Fully agreed. Thanks for the snippet!